sourmash/lca/command_index.py

```diff
@@ -178,77 +213,120 @@ def index(args):
     else:
         inp_files = list(args.signatures)
 
+    #
+    # main loop, connecting lineage ID to signature.
+    #
+
     n = 0
     total_n = len(inp_files)
     record_duplicates = set()
     record_no_lineage = set()
-    record_remnants = set(assignments_idx.keys())
+    record_remnants = set(ident_to_idx.keys())
+    record_used_lineages = set()
+    record_used_idents = set()
     for filename in inp_files:
         n += 1
         for sig in load_signatures(filename, ksize=args.ksize):
             notify(u'\r\033[K', end=u'')
             notify('... loading signature {} (file {} of {})', sig.name()[:30], n, total_n, end='\r')
             debug(filename, sig.name())
 
-            if sig.md5sum() in md5_to_lineage:
+            if sig.md5sum() in md5_to_name:
                 notify('\nWARNING: in file {}, duplicate md5sum: {}; skipping', filename, sig.md5sum())
                 record_duplicates.add(filename)
                 continue
 
-            name = sig.name()
+            ident = sig.name()
             if args.split_identifiers: # hack for NCBI-style names, etc.
-                name = name.split(' ')[0].split('.')[0]
-
-            # is this one for which we have a lineage assigned?
-            lineage_idx = assignments_idx.get(name)
-            if lineage_idx is None:
-               notify('\nWARNING: no lineage assignment for {}.', name)
-               record_no_lineage.add(name)
+                ident = ident.split(' ')[0].split('.')[0]
+
+            # store full name
+            ident_to_name[ident] = sig.name()
+
+            # store md5 -> name too
+            md5_to_name[sig.md5sum()] = sig.name()
+            
+            # remove from our list of remnant lineages
+            try:
+                record_remnants.remove(ident)
+            except KeyError:
+                # @CTB
```

Comment:
Hmm, is this a new pattern? If you `pass` an exception, you need to sign it to be made responsible at some point in the future? =]